socorro/cron/jobs/archivescraper.py

```diff
@@ -316,137 +254,272 @@ def scrape_candidates(self, product_name, archive_directory):
-                            # insert two entries--one for the last rc in the beta channel
-                            # and one for the final release in the release channel. This
-                            # makes it possible to look up version strings for beta and rc
-                            # builds in one request.
-
-                            # Insert the rc beta build
-                            data['release_channel'] = 'beta'
-                            data['version_string'] = rc_version_string
-                            self.insert_build(**data)
-
-                            # Insert the final release build
-                            data['version_string'] = version_root
-                            data['release_channel'] = 'release'
-                            self.insert_build(**data)
-
-                        else:
-                            # This is the final build for a beta release, so we insert both
-                            # an rc as well as a final as betas
-                            data['version_string'] = version_root
-                            self.insert_build(**data)
-
-                            data['version_string'] = rc_version_string
-                            self.insert_build(**data)
+                        # This is the final build for a beta release, so we insert both
+                        # an rc as well as a final as betas
+                        data['version_string'] = version_root
+                        build_data.append(data)
+
+                        second_data = copy.deepcopy(data)
+                        second_data['version_string'] = rc_version_string
+                        build_data.append(second_data)
+
+                else:
+                    if data['release_channel'] == 'release':
+                        # This is a release channel build, but it's not a final build,
+                        # so insert it as an rc beta build
+                        data['version_string'] = rc_version_string
+                        data['release_channel'] = 'beta'
+                        build_data.append(data)
 
                     else:
-                        if data['release_channel'] == 'release':
-                            # This is a release channel build, but it's not a final build,
-                            # so insert it as an rc beta build
-                            data['version_string'] = rc_version_string
-                            data['release_channel'] = 'beta'
-                            self.insert_build(**data)
-
-                        else:
-                            # Insert the rc beta build
-                            data['version_string'] = rc_version_string
-                            self.insert_build(**data)
+                        # Insert the rc beta build
+                        data['version_string'] = rc_version_string
+                        build_data.append(data)
+
+        return build_data
```

Comment:
This method used to insert the builds. Now it returns a list of all the builds to be inserted.